pkg/agent/agent_test.go

```diff
@@ -109,6 +114,91 @@ kind: Config`)
+		i++
+	}
+
+	err := aCtrl.CreateAddOnPlacementScore(ctx)
+	assert.Nil(t, err, "err nil when CreateAddOnPlacementScore was successfull")
+
+	clusterClaim, err := aCtrl.spokeClustersClient.ClusterV1alpha1().ClusterClaims().Get(context.TODO(), hostedClusterCountFullClusterClaimKey, metav1.GetOptions{})
+	assert.Nil(t, err, "is nil when the clusterclaim is found")
+	assert.Equal(t, strconv.FormatBool(false), clusterClaim.Spec.Value)
+
+	placementScore := &clusterv1alpha1.AddOnPlacementScore{}
+	placementScoreNN := types.NamespacedName{Name: util.HostedClusterScoresResourceName, Namespace: aCtrl.clusterName}
+	err = aCtrl.hubClient.Get(ctx, placementScoreNN, placementScore)
+	assert.Nil(t, err, "is nil when addonPlacementScore is found")
+	assert.Equal(t, int32(util.MaxHostedClusterCount-1), placementScore.Status.Scores[0].Value)
+
+	// Create one more hosted cluster and expect the cluserclaim to have hostedclustercount.full.hypershift.openshift.io=true
+	// indicating it reached the maximum number of hosted clusters
+	hc := getHostedCluster(hcNN)
+	hc.SetName("test-80")
+	err = aCtrl.hubClient.Create(ctx, hc)
+	assert.Nil(t, err, "err nil when hosted cluster is created successfull")
+
+	err = aCtrl.CreateAddOnPlacementScore(ctx)
+	assert.Nil(t, err, "err nil when CreateAddOnPlacementScore was successfull")
+
+	clusterClaim, err = aCtrl.spokeClustersClient.ClusterV1alpha1().ClusterClaims().Get(context.TODO(), hostedClusterCountFullClusterClaimKey, metav1.GetOptions{})
+	assert.Nil(t, err, "is nil when the clusterclaim is found")
+	assert.Equal(t, strconv.FormatBool(true), clusterClaim.Spec.Value)
+
+	err = aCtrl.hubClient.Get(ctx, placementScoreNN, placementScore)
+	assert.Nil(t, err, "is nil when addonPlacementScore is found")
+	assert.Equal(t, int32(util.MaxHostedClusterCount), placementScore.Status.Scores[0].Value)
+
+	// Delete one hosted cluster and expect the cluserclaim to have hostedclustercount.full.hypershift.openshift.io=false
+	// indicating it did not reach the maximum number of hosted clusters after removing one
+	err = aCtrl.hubClient.Delete(ctx, hc)
+	assert.Nil(t, err, "err nil when hosted cluster is deleted successfull")
+
+	err = aCtrl.CreateAddOnPlacementScore(ctx)
+	assert.Nil(t, err, "err nil when CreateAddOnPlacementScore was successfull")
+
+	clusterClaim, err = aCtrl.spokeClustersClient.ClusterV1alpha1().ClusterClaims().Get(context.TODO(), hostedClusterCountFullClusterClaimKey, metav1.GetOptions{})
+	assert.Nil(t, err, "is nil when the clusterclaim is found")
+	assert.Equal(t, strconv.FormatBool(false), clusterClaim.Spec.Value)
+
+	err = aCtrl.hubClient.Get(ctx, placementScoreNN, placementScore)
+	assert.Nil(t, err, "is nil when addonPlacementScore is found")
+	assert.Equal(t, int32(util.MaxHostedClusterCount-1), placementScore.Status.Scores[0].Value)
+
+	// Simulate that it fails to get a list of all hosted clusters. In this case, the addonPlacementScore
```

Comment:
nit: I wouldn't mind seeing this as its own test.

Reply:
Done Thank you